### PR TITLE
Quick review of 'create framework' tutorial

### PR DESCRIPTION
**Pull request by wouterj**

### CODE REVIEW

File: create_framework/routing.rst [hunk 1]
```diff
@@ -43,13 +41,8 @@ One very important aspect of any website is the form of its URLs. Thanks to
...
 associated response, but it is not yet flexible enough. For instance, we might
 want to support dynamic paths to allow embedding data directly into the URL
-instead of relying on a query string:
-
-    # Before
-    /hello?name=Fabien
-
-    # After
-    /hello/Fabien
+instead of relying on a query string (like ``/hello/Fabien`` instead of
```
Comment (javiereguiluz) [inline create_framework/routing.rst]: I'm not sure if the new version is much better than the old one. The repeated `instead of` is confusing to me.

File: create_framework/http-kernel-httpkernelinterface.rst [hunk 2]
```diff
@@ -8,12 +8,18 @@ goal by making our framework implement ``HttpKernelInterface``::
...
 
+    // ...
+
     interface HttpKernelInterface
     {
         /**
          * @return Response A Response instance
          */
-        function handle(Request $request, $type = self::MASTER_REQUEST, $catch = true);
+        public function handle(
```
Comment (javiereguiluz) [inline create_framework/http-kernel-httpkernelinterface.rst]: I'm aware of the line-length rule, but this kind of changes look awful to me. I'd prefer to keep the method declarations in one line, unless they have lots of long arguments.

File: create_framework/http_kernel_httpkernelinterface.rst [hunk 3]
```diff
@@ -8,12 +8,18 @@ goal by making our framework implement ``HttpKernelInterface``::
 
     namespace Symfony\Component\HttpKernel;
 
+    // ...
+
```
Comment (xabbuh) [inline create_framework/http_kernel_httpkernelinterface.rst]: You don't need the blank line, do you?

File: create_framework/separation_of_concerns.rst [hunk 4]
```diff
@@ -79,14 +77,10 @@ To wrap up the refactoring, let's move everything but routes definition from
...
 
     {
-        "require": {
-            "symfony/http-foundation": "2.5.*",
-            "symfony/routing": "2.5.*",
-            "symfony/http-kernel": "2.5.*"
-        },
+        "...": "...",
         "autoload": {
             "psr-0": { "Simplex\\": "src/", "Calendar\\": "src/" }
```
Comment (xabbuh) [inline create_framework/separation_of_concerns.rst]: I suggest to switch to PSR-4 here.

**Comment by xabbuh** [comment]
You could also add the missing use statement for the HttpKernel component (see https://github.com/symfony/symfony-docs/pull/5556#discussion_r35374644).

**Comment by WouterJ** [comment]
That was the original reason for doing this PR, but it turns out it isn't missing. It's included in a previous chapter or code example.

**Comment by xabbuh** [comment]
I see, but then we should imo add `// ...` placeholders where they are needed in that file.

**Comment by WouterJ** [comment]
Then we need to do it for 99% of the examples (as it's a tutorial, all code examples are an improvement of the previous shown example)

**Comment by xabbuh** [comment]
Hm, you're right. I think we can keep it as is then.

File: create_framework/unit_testing.rst [hunk 5]
```diff
@@ -14,17 +14,12 @@ using `PHPUnit`_. Create a PHPUnit configuration file in
...
-             convertErrorsToExceptions="true"
-             convertNoticesToExceptions="true"
-             convertWarningsToExceptions="true"
-             processIsolation="false"
-             stopOnFailure="false"
-             syntaxCheck="false"
-             bootstrap="vendor/autoload.php"
+    <phpunit
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/4.1/phpunit.xsd"
```
Comment (stof) [inline create_framework/unit_testing.rst]: I suggest using a newer version than 4.1, which is EOLed

File: redirection_map [hunk 6]
```diff
@@ -26,3 +26,12 @@
...
 /cookbook/security/voters_data_permission /cookbook/security/voters
+/create_framework/http_foundation /create_framework/http-foundation
+/create_framework/front_controller /create_framework/front-controller
+/create_framework/http_kernel_controller_resolver /create_framework/http-kernel-controller-resolver
+/create_framework/separation_of_concerns /create_framework/separation-of-concerns
+/create_framework/unit_testing /create_framework/unit-testing
+/create_framework/event_dispatcher /create_framework/event-dispatcher
+/create_framework/http_kernel_httpkernelinterface /create_framework/http-kernel-httpkernelinterface
+/create_framework/http_kernel_httpkernel_class /create_framework/http-kernel-httpkernel-class
+/create_framework/dependency_injection /create_framework/dependency-injection
```
Comment (stof) [inline redirection_map]: the redirection map is wrong here. You swapped the old and new URLs. The URLs with an underscore are the new ones

File: create_framework/http-kernel-httpkernelinterface.rst (hunk 2 above)
Comment (OskarStark) [inline create_framework/http-kernel-httpkernelinterface.rst]: me too, but adding the ```public``` is :+1:

File: create_framework/separation_of_concerns.rst (hunk 4 above)
Comment (OskarStark) [inline create_framework/separation_of_concerns.rst]: me, to, as the current symfony installer use PSR-4:

```
    "autoload": {
        "psr-4": {
            "": "src/",
            "SymfonyStandard\\": "app/SymfonyStandard/"
        }
    },
```

File: redirection_map (hunk 6 above)
Comment (OskarStark) [inline redirection_map]: should we add some tests, which urls are callable?

(this comment is not directly related to this PR, sorry)

**Comment by WouterJ** [comment]
PR updated and ready imo

**Comment by weaverryan** [comment]
Thanks Wouter! The only thing I wonder about (and it was done this way before), is if we're going to use full namespaces for classes instead of use statements, why not prefix them with their opening \ so that they'll work even if someone copies some of this code into a namespaced file. What do you think about that?

File: create_framework/routing.rst (hunk 1 above)
Comment (weaverryan) [inline create_framework/routing.rst]: I tweaked this here to avoid the 2 `instead of` - sha: 3640b2a13ba74bdafb8b64b496ef988a8a12ee64
